modules/extension/xsd/xsd-core/src/test/java/org/geotools/xsd/ParserTest.java

```diff
@@ -343,4 +345,48 @@ public InputSource resolveEntity(
         parser.validate(
                 MLSchemaLocationResolver.class.getResourceAsStream("mails-external-entities.xml"));
     }
+
+    /** Tests entity expansion limit configuration on Parser. */
+    @Test
+    public void testEntityExpansionLimit() throws Exception {
```

Comment:
Maybe also have a test showing that an expansion below the set limits works

Reply:
Indeed makes sense, added.